PR: https://github.com/odoo/odoo/pull/127616

From: 3bd841086d92246e55771785bcb69501af712ab6
From: JF Aubert
Date: 2023-10-18 00:44:52

Structural Changes: 6
Total Changes: 18

[IMP] maintenance,mrp: add worksheet templates

Manage worksheet templates on maintenance requests.

Also add 'Notes' and 'Instruction' tabs in form view.

closes odoo/odoo#127616

Task: 3186753
Related: odoo/enterprise#43763
Signed-off-by: Quentin Wolfs (quwo) <quwo@odoo.com>

================================= pseudo patch: =================================

--- a/addons/maintenance/models/__init__.py
+++ b/addons/maintenance/models/__init__.py
@@ -1,3 +1,4 @@
 # -*- coding: utf-8 -*-
 
 from . import maintenance
+from . import res_config_settings

--- a/addons/maintenance/models/maintenance.py
+++ b/addons/maintenance/models/maintenance.py
@@ -306,6 +306,13 @@ class MaintenanceRequest(models.Model):
     maintenance_team_id = fields.Many2one('maintenance.team', string='Team', required=True, default=_get_default_team_id, check_company=True)
     duration = fields.Float(help="Duration in hours.")
     done = fields.Boolean(related='stage_id.done')
+    instruction_type = fields.Selection([
+        ('pdf', 'PDF'), ('google_slide', 'Google Slide'), ('text', 'Text')],
+        string="Instruction", default="text"
+    )
+    instruction_pdf = fields.Binary('PDF')
+    instruction_google_slide = fields.Char('Google Slide', help="Paste the url of your Google Slide. Make sure the access to the document is public.")
+    instruction_text = fields.Html('Text')
 
     def archive_equipment_request(self):
         self.write({'archive': True})

--- a/None
+++ b/addons/maintenance/models/res_config_settings.py
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    module_maintenance_worksheet = fields.Boolean(string="Custom Maintenance Worksheets")
